python/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -324,17 +324,18 @@ class RandomForestClassifier(Base):
         cdef size_t params_t64
         if self.n_cols:
             # only if model has been fit previously
-            model_pbuf_bytes = self._get_protobuf_bytes()
-            params_t = <uintptr_t> self.rf_forest
-            rf_forest = \
-                <RandomForestMetaData[float, int]*>params_t
-            params_t64 = <uintptr_t> self.rf_forest64
-            rf_forest64 = \
-                <RandomForestMetaData[double, int]*>params_t64
-            state["rf_params"] = rf_forest.rf_params
-            state["rf_params64"] = rf_forest64.rf_params
-        else:
-            model_pbuf_bytes = bytearray()
+            self._get_protobuf_bytes()  # Ensure we have this cached
+            if self.rf_forest:
+                params_t = <uintptr_t> self.rf_forest
```

Comment:
I was in the process of writing a note about having both of these always assigned to an object and just noticed you added the `if` statements in here. This is much better. Would be nice if we could somehow push more of the typing down to the c++ layer since cython seems so limited.